libs/output-mapping/src/Storage/TableStructureValidator.php

```diff
@@ -133,11 +134,25 @@ private function validateTypedTable(array $table, array $schemaColumns, string $
             try {
                 $schemaColumnType = $schemaColumn->getDataType()->getBackendTypeName($bucketBackend);
                 $tableColumnType = $tableColumn['definition']['type'];
+                $columnDefinition = new $columnDefinitionClassName($schemaColumnType);
+                if (method_exists($columnDefinition, 'getBackendBasetype')) {
+                    $schemaColumnType = $columnDefinition->getBackendBasetype();
```

Comment:
Ty já nevím, opravdu to dělá to, co to má dělat? Jako název tý metody je dost divnej, zdá se, že dělá to co chceme, ale nemusíme pak u některých typů zahodit délku ?

Reply:
název - tak získává základní typ na backendu z aliasu :)

dělá co má dělat - jo dělá :) ř. 137 vytvoří classu z daným sloupečkem a tady pokud je to alias tak to přeloží na základní typ (např NVARCHAR2 -> VARCHAR)

délka - ta by se měla zachovat, však když pošlu třeba ten NVARCHAR2(255) tak ve snflk se to založí stejně jako VARCHAR(255) a tak to i musím porovnávat
----
snad jsem odpověděl na vše :)

Reply:
https://docs.snowflake.com/en/sql-reference/data-types-numeric#int-integer-bigint-smallint-tinyint-byteint

INT , INTEGER , BIGINT , SMALLINT , TINYINT , BYTEINT
Synonymous with NUMBER, except that precision and scale cannot be specified

Reply:
no do toho stavu že pošleš INTEGER s length tak bychom se neměli vůbec dostat, protože to vyhučí pak v connection při zakládání tabulky 

```
Invalid request:\n - columns[0][definition][length]: \"'22' is not valid length for INTEGER\"
```

Reply:
a tohle je jen validator, takže tady se sice INTEGER přeloží na NUMBER ale tabulka se zakládá stejně s INTEGERem